ddtrace/tracer/sampler.go

```diff
@@ -24,6 +24,36 @@ import (
 	"golang.org/x/time/rate"
 )
 
+// samplerName specifies the identification of a sampler which was
+// responsible for a certain sampling decision.
+type samplerName int8
+
+const (
+	// samplerNone specifies that the sampling decision was already made
+	// and it's noop except propagating the decision.
+	samplerNone samplerName = math.MinInt8
+	// samplerUnknown specifies that the span was sampled
+	// but, the tracer was unable to identify the sampler.
+	samplerUnknown samplerName = -1
+	// samplerDefault specifies that the span was sampled without any sampler.
```

Comment:
How is this different from `samplerUnknown`? These 3 are still confusing and the documentation doesn't clarify the confusion.

Reply:
This was supposed to be used until the sampling rates come from the agent for the first time.
But, I didn't implement that logic in this PR. We can do it later.
I can remove `samplerDefault` from this PR now, then add back when we implement that logic.